datadog/resource_datadog_service_account.go

```diff
@@ -0,0 +1,280 @@
+package datadog
+
+import (
+	"context"
+	"log"
+
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceDatadogServiceAccount() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a Datadog service account resource. This can be used to create and manage Datadog service accounts.",
+		CreateContext: resourceDatadogServiceAccountCreate,
+		ReadContext:   resourceDatadogServiceAccountRead,
+		UpdateContext: resourceDatadogServiceAccountUpdate,
+		DeleteContext: resourceDatadogUserDelete,
```

Comment:
```suggestion
		DeleteContext: resourceDatadogServiceAccountDelete,
```

Reply:
sorry this was leftover code - there is no functional difference between deleting a user resource versus a service account resource, so it should be `resourceDatadogUserDelete`